### PR TITLE
Fixed a backup bug

### PR DESCRIPTION
**Pull request by monadbobo**
Fixed an infinite loop that could occur when backing up only part of a space.

jira.nebula-graph.io/projects/NG/issues/NG-377

### CODE REVIEW

**Review by critical27 — APPROVED** [review]

**Comment by critical27** [comment]
Do we have known bug in br tool after these PR merged?

File: src/meta/processors/admin/SnapShot.cpp [hunk 1]
```diff
@@ -92,7 +92,7 @@ StatusOr<std::map<GraphSpaceID, std::set<HostAddr>>> Snapshot::getSpacesHosts()
         LOG(ERROR) << "Get hosts meta data error";
         return Status::Error("Get hosts meta data error");
     }
-    while (iter->valid()) {
+    for (; iter->valid(); iter->next()) {
```
Comment (bright-starry-sky) [inline src/meta/processors/admin/SnapShot.cpp:95]: What's the difference? I don't understand.

Comment (critical27) [reply src/meta/processors/admin/SnapShot.cpp:95]: the continue will loop forever

**Comment by monadbobo** [comment]
> Do we have known bug in br tool after these PR merged?

#277 #247 #283

**Review by bright-starry-sky — APPROVED** [review]
Nice job. LGTM.

**Review by critical27 — APPROVED** [review]

**Pull request merged**